app/models/docker_parameter.rb

```diff
@@ -0,0 +1,21 @@
+class DockerParameter < ActiveRecord::Base
+  extend FriendlyId
+  friendly_id :key
+  include Parameterizable::ByIdName
+
+  validates_lengths_from_database
+
+  include Authorizable
+  validates :key, :presence => true, :no_whitespace => true
```

Comment:
I wonder if all the validations in the concern could've been substituted by a validation of uniqueness with scope here?